service/history/historyEngine.go

```diff
@@ -2177,7 +2177,10 @@ func (e *historyEngineImpl) RecordChildExecutionCompleted(
 				_, err = mutableState.AddChildWorkflowExecutionCanceledEvent(parentInitiatedID, completedExecution, attributes)
 			case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED:
 				attributes := completionEvent.GetWorkflowExecutionTerminatedEventAttributes()
-				_, err = mutableState.AddChildWorkflowExecutionTerminatedEvent(parentInitiatedID, completedExecution, attributes)
+				// Resetter check
+				if attributes.Identity != "resetter" {
```

Comment:
This check should live in the caller side I think (i.e. the logic for [processing workflow close transfer task](https://github.com/temporalio/temporal/blob/master/service/history/transferQueueActiveTaskExecutor.go#L357)) so that we won't have the rpc call at all. There's already some logic for determining if reporting to parent is needed or not and they should be grouped together. 

The [standby task execution logic](https://github.com/temporalio/temporal/blob/3dbabab407bb069f8c8cb219344b4e2195ffda5c/service/history/transferQueueStandbyTaskExecutor.go#L278) also needs to be updated so that it this case standby child workflow won't try to verify its result is recorded. Probably created a util function that's shared by both active and standby logic.